src/MediatR/Mediator.cs

```diff
@@ -73,20 +88,65 @@ public void Publish(INotification notification)
 
         public Task PublishAsync(IAsyncNotification notification)
         {

```

Comment:
I see. This would be an issue for me as well, will have to update every call site. So settings behavior in one place is a good idea.